src/main/java/org/checkerframework/specimin/UnsolvedSymbolVisitor.java

```diff
@@ -293,13 +300,22 @@ public Visitable visit(ExplicitConstructorInvocationStmt node, Void arg) {
 
   @Override
   public Visitable visit(NameExpr node, Void arg) {
+    //  This method explicitly handles NameExpr instances that represent fields of classes but are
+    // not explicitly shown in the code. For example, if "number" is a field of a class, then
+    // "return number;" is an expression that this method will address. If the NameExpr instance is
+    // not a field of any class, or if it is a field of a class but is explicitly referenced, such
+    // as "Math.number," we handle it in other visit methods.
```

Comment:
What if `number` is a local variable, as in:
```
int number = 5;
return number;
```

I don't see anything in this code that accounts for that case. Can you add a test case to make sure no synthetic super class field is generated for locals?